agent/engine/task_manager.go

```diff
@@ -504,29 +504,39 @@ func (mtask *managedTask) emitResourceChange(change resourceStateChange) {
 func (mtask *managedTask) emitTaskEvent(task *apitask.Task, reason string) {
 	event, err := api.NewTaskStateChangeEvent(task, reason)
 	if err != nil {
-		seelog.Infof("Managed task [%s]: unable to create task state change event [%s]: %v",
+		if _, ok := err.(api.SkippableStateChange); ok {
+			seelog.Debugf("Managed task [%s]: skipping emitting event for task [%s]: %v",
+				task.Arn, reason, err)
+			return
+		}
+		seelog.Errorf("Managed task [%s]: error emitting event for task [%s]: %v",
 			task.Arn, reason, err)
 		return
 	}
 	seelog.Infof("Managed task [%s]: sending task change event [%s]", mtask.Arn, event.String())
 	mtask.stateChangeEvents <- event
-	seelog.Infof("Managed task [%s]: sent task change event [%s]", mtask.Arn, event.String())
+	seelog.Debugf("Managed task [%s]: sent task change event [%s]", mtask.Arn, event.String())
```

Comment:
i'd prefer this and line 539 stay at info level so that if we somehow miss sending some state changes, we know the problem is on the task manager side or the handler that consumes stateChangeEvents channel (many people don't turn on debug logs by default)

Reply:
okay, I can leave this at info